Assets/MixedRealityToolkit.Examples/Experimental/SceneUnderstanding/Scripts/DemoSceneUnderstandingController.cs

```diff
@@ -0,0 +1,395 @@
+            platformToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Platform);
+            wallToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Wall);
+            floorToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Floor);
+            ceilingToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Ceiling);
+            worldToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.World);
+            completelyToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.CompletelyInferred);
+            backgroundToggle.GetComponent<Interactable>().IsToggled = observer.SurfaceTypes.HasFlag(SpatialAwarenessSurfaceTypes.Background);
+            usePersistentToggle.GetComponent<Interactable>().IsToggled = observer.UsePersistentObjects;
+        }
+
+        public void OnObservationAdded(MixedRealitySpatialAwarenessEventData<SpatialAwarenessSceneObject> eventData)
+        {
+            // This method called everytime a SceneObject created by the SU observer
+            // The eventData contains everything you need do something useful
+
+            var sceneObject = eventData.SpatialObject; // alias
+
+            if (sceneObject.SurfaceType == SpatialAwarenessSurfaceTypes.Platform)
+            {
+                platforms.Add(sceneObject);
+            }
+
+            if (InstantiatePrefabs)
+            {
+                var prefab = Instantiate(InstantiatedPrefab);
+                prefab.transform.SetPositionAndRotation(sceneObject.Position, sceneObject.Rotation);
+
+                if (InstantiatedParent)
+                {
+                    prefab.transform.SetParent(InstantiatedParent);
+                }
+
+                // A prefab can implement the ISpatialAwarenessSceneObjectConsumer contract
+                // this will let the prefab author decide how it wants to "react" to the new sceneObject
+                // In the demo scene, the prefab will scale itself to fit quad extents
+
+                foreach (var x in prefab.GetComponents<ISceneUnderstandingSceneObjectConsumer>())
+                {
+                    x.OnSpatialAwarenessSceneObjectCreated(sceneObject);
+                }
+            }
+        }
+
+        public async void PutStuffOnNearestPlatform()
+        {
+            var platformCount = platforms.Count;
+
+            if (platformCount < 1)
+            {
+                await Task.Yield();
+                return;
+            }
+
+            // Find the Guid of our nearest neighbor
+
+            float closestDistance = float.MaxValue;
+            Guid closestGuid;
+            bool foundGuid = false;
```

Comment:
nit: foundGuid doesn't seem necessary, can't you just check closestGuid != empty/default?